rct229/rulesets/ashrae9012019/section19/section19rule4.py

```diff
@@ -0,0 +1,277 @@
+                        self,
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, False),
+                        each_rule=Section19Rule4.RuleSetModelInstanceRule.BuildingSegmentRule.ZoneRule.SpaceRule(),
+                        index_rmr="baseline",
+                        list_path="$.spaces[*]",
+                        required_fields={
+                            "$": ["infiltration"],
+                        },
+                    )
+
+                def create_data(self, context, data):
+                    day_of_week_for_january_1 = data["day_of_week_for_january_1"]
+                    zone_b = context.baseline
+
+                    # check infiltration
+                    inf_pass_cooling = True
+                    multiplier_schedule_infiltration = zone_b["infiltration"][
+                        "multiplier_schedule"
+                    ]
+                    multiplier_sch = data[multiplier_schedule_infiltration]
+                    hourly_values = multiplier_sch["hourly_values"]
+
+                    most_used_weekday_hourly_schedule = (
+                        get_most_used_weekday_hourly_schedule(
+                            hourly_values, day_of_week_for_january_1
+                        )
+                    )
+                    for x in range(0, int(len(hourly_values) / 24), 24):
+                        if (
+                            most_used_weekday_hourly_schedule
+                            != hourly_values[x : x + 24]
+                        ):
+                            inf_pass_cooling = False
+                            break
+
+                    return {"inf_pass_cooling": inf_pass_cooling}
+
+                class SpaceRule(RuleDefinitionBase):
+                    def __init__(self):
+                        super(
+                            Section19Rule4.RuleSetModelInstanceRule.BuildingSegmentRule.ZoneRule.SpaceRule,
+                            self,
+                        ).__init__(
+                            rmrs_used=UserBaselineProposedVals(False, True, False),
+                            required_fields={
+                                "$": ["lighting_space_type", "ventilation_space_type"],
+                            },
+                            manual_check_required_msg=UNDETERMINED_MSG,
+                            fail_msg=FAIL_MSG,
+                        )
+
+                    def are_two_schedules_same_helper(
+                        self, hourly_values, most_used_weekday_hourly_schedule
+                    ):
+                        for x in range(0, int(len(hourly_values) / 24), 24):
+                            if (
+                                most_used_weekday_hourly_schedule
+                                != hourly_values[x : x + 24]
+                            ):
+                                return False
+                        return True
```

Comment:
You don't need this helper function anymore, the `design_cooling_multiplier_sch` is a 24 hours schedule and the `most_used_weekday_hourly_schedule` is a 24 hours schedule.
Since the value and order matters in this comparison, you can just do
`most_used_weekday_hourly_schedule == design_cooling_multiplier_sch`

Reply:
Good point. Addressed.